tests/impl.cpp

```diff
@@ -129,6 +129,36 @@ class SSE2NEONTestImpl : public SSE2NEONTest
                 }
             }
 
+#if SSE2NEON_PRECISE_MINMAX
+            if (test == it_mm_max_ps || test == it_mm_max_ss ||
+                test == it_mm_min_ps || test == it_mm_min_ss) {
+                // Make sure the NaN values are included in the testing
+                // one out of four times.
+                if ((rand() & 3) == 0) {
+                    uint32_t r1 = rand() & 3;
+                    uint32_t r2 = rand() & 3;
```

Comment:
What if `r1` equals to `r2`? Would you enforce different indexes?

Reply:
The indexes can be the same or different since the situation of both operands are `NaN` should be tested as well.